main.py

```diff
@@ -46,39 +47,148 @@
+    stored into a main_metadata.json file.
+   
+    Note:  function asserts that cmd_name must be filled out 
+    for the command_policy option and if an input file is 
+    specified then it must exist else the program will exit.  
 
   <Arguments>
-    None.
+    args:
+      The arguments that are passed from the command line.
 
   <Exceptions>
-    TBD.
+    IOError: 
+      Returned if the input file does not exist.
 
-  <Return>
+  <Returns>
     None.
   """
 
-  # Setup the command line parser
-  args = get_command_line_args()
+  commands = []
+  # Setup the command policy here.  The user can either pass
+  # a file in from the commandline (command_policy option), 
+  # otherwise pass a single command line entry.
+  if args.command_policy:
+    # Begin processing the commands from the command policy. 
+    command_policy_filepath = args.command_policy
+    commandList = tuf.util.load_json_file(command_policy_filepath)
+    commands = commandList["commands"]
+  else:
+    # The command is read from the commandline (COMMAND var).
+    # We will autoname the file generic_metadata.json. 
+    tmp_commands = [[args.command, None, "generic", args.input]]
+    commands = json.dumps(tmp_commands)
+    commands = tuf.util.load_json_string(commands)
+
+
+  # Begin to verify and execute the commands.
+  cumulative_metadata_hash = ""
+  metadata_files_processed = dict()
+  sha256_hasher = hashlib.sha256()
+
+  # Begin to iterate through the commands.  First verify 
+  # the command setup is valid, next process the actual command.
+  for cmd_elem in commands:
+    cmd_name = cmd_elem[2]
+    # Check if the cmd_name is empty or is 'main'.  We need to use
+    # the cmd_name field when naming our metadata files so 
+    # make sure they are set here.
+    assert (cmd_name), "Error: The cmd_name cannot be null."
+    assert (cmd_name != "main"), "Error: The cmd_name cannot equal 'main'."
+    # Validate the input file exists else flag an error.
+    if (cmd_elem[3]):
+      try:
+        os.path.exists(cmd_elem[3])
+      except IOError:
+        print "Error:  The input file was specified, however it does not exist."
+
+
+    # Process the commands and create the metadata files here.
+    hash_metadata, metadata_filepath = process_command(cmd_elem, args.policy)
+    metadata_files_processed[cmd_name + "_metadata_path"] = metadata_filepath
+    metadata_files_processed[cmd_name + "_metadata_hash"] = hash_metadata
+
+    # Add to the hasher to calculate the cumulative hash of metadata files.
+    sha256_hasher.update(hash_metadata)
+    metadata_files_processed["cumulative_metadata_hash"] = sha256_hasher.hexdigest()
+    metadata_files_processed["timestamp"] = str(datetime.datetime.utcnow())
+    
+    # Sign the data and write to main_metadata.json.
+    signed_metadata = signing.sign_json(metadata_files_processed)
+    main_metadata_path = os.path.join(DEFAULT_WORK_DIR, "main_metadata")
```

Comment:
Ok, and what about tracing the hashes of tarballs? That isn't implemented in this design, correct?